packages/components/src/Drawer/Drawer.component.js

```diff
@@ -182,12 +205,27 @@ function Drawer({
 	tabs,
 	withTransition,
 	getComponent,
+	selectedTabKey,
 }) {
 	if (!children) {
 		return null;
 	}
 
 	const TabBarComponent = Inject.get(getComponent, 'TabBar', TabBar);
+
+	let activeTab = {};
+	let activeTabItem = [];
+	let customTabs;
+	if (tabs && tabs.items.length > 0) {
+		customTabs = Object.assign({}, tabs);
+
+		if (selectedTabKey) {
+			Object.assign(customTabs, { selectedKey: selectedTabKey });
+			activeTab = tabs.items.find(tab => tab.key === selectedTabKey);
+		}
+		activeTabItem = get(activeTab, 'footerActions', {});
+	}
```

Comment:
To be honest, I don't really understand why this huge if instead of passing undefined values to functions that already has default parameters. For example, I think that something like that could do the same job with less code and complexity. What do you think ? : 

```
	const activeTab = tabs.items.find(tab => tab.key === selectedTabKey);
	const activeTabItem = get(activeTab, 'footerActions', {});

	[...]

	<TabBarComponent
		{...tabs}
		selectedKey=selectedTabKey
		className={classnames('tc-drawer-tabs', theme['tc-drawer-tabs'])}
	/>

	[...]

	<ActionBar
		{...combinedFooterActions(onCancelAction, footerActions, activeTabItem)}
		className={classnames('tc-drawer-actionbar', theme['tc-drawer-actionbar'])}
	/>
```